### PR TITLE
Revamp fix_packages

### LINKED ISSUE #57: fix_packages.php doesn't work on PHP >= 8.0
The fix_packages.php tool doesn't work on PHP 8.0 or higher due to its reliance on the create_function() function.

### LINKED ISSUE #62: fix_packages needs some tlc - deprecated create_function
Deprecated: Function create_function() is deprecated in \yada\fix_packages.php on line 189

Deprecated: Function create_function() is deprecated in \yada\fix_packages.php on line 198

### PR DESCRIPTION
**Pull request by jdarwood007**
Fixes #57
Fixes #62

### CODE REVIEW

**Comment by jdarwood007** [comment]
FYI I only tested this on 2.0 and 2.1.  Didn't test 1.x

**Comment by sbulen** [comment]
I don't recall seeing this request for review earlier.  Sorry about that.

I'll take a look.  I am not actually familiar with this tool or what it does...  The one time I tried to figure it out it didn't work with php 8...  

I thought this wasn't necessary for 2.1???   And, to be honest, I don't want to spend time on anything specific to 2.0, which is in security fix only mode.  I'd like to focus all bandwidth on 2.1 going forward.

**Comment by DiegoAndresCortes** [comment]
Some users were trying to use it on 2.1 since it’s valid, but reported it as faulty using php 8
Perhaps Sleepy forgot about this one too.

**Comment by sbulen** [comment]
Can someone tell me what this is supposed to do?  😆

Is there a description anywhere?  It's not in the code...

I *think* it's supposed to help you:
 - Flag packages as installed or uninstalled
 - Activate or inactivate integration hooks

Anything else?

I actually don't have any 2.0 environments to test anymore...  _I thought I was done with 2.0..._  I'm trying to build a 2.0.19 env to test this...

**Comment by jdarwood007** [comment]
I believe that is what it was designed to do.  Fix a broken package where your forum may not otherwise work.

**Comment by sbulen** [comment]
OK, just pulled it up & I suspect a problem.  I have 5 packages installed in my test environment.  When I pull up the old version, it lists the 5 packages.  When I pull up the new version, it lists 0 packages.  No errors...

**_Screenshot 1 - old version (php 7.4):_**
old_version

**_Screenshot 2 - new version (php 8.0):_**
none_show

**_Screenshot 3 - 5 packages installed in pacman:_**
packages_installed

**Comment by sbulen** [comment]
There is a definite issue with how it is calling the list builder.  The param is not getting passed properly...  isInstalled is false, so it's looking for packages with an installed value of 0.  

In my simple test, they all have a value of 1, so no packages are returned.

**Review by sbulen — CHANGES_REQUESTED** [review]
Needs some tweakin'.

The list builder parameters aren't being passed properly, so packages aren't being displayed.

**Comment by jdarwood007** [comment]
I will try to get around to it soon as I can.

File: fix_packages.php [hunk 1]
```diff
@@ -40,80 +36,230 @@
...
+						<input type="submit" name="remove_packages" value="' . $this->lang('pack_button_' . (!empty($this->isInstalling) ? 'remove' : 'install')) . '" class="button_submit" />
+						<input type="submit" name="remove_hooks" value="' . $this->lang('remove_hooks') . '" class="button_submit" />',
+					'class' => 'righttext',
+				]
+			],
+		];
+	}
 
-	if (!empty($_POST['remove']) && is_array($_POST['remove']))
+	public static function list_getPacks(bool $isInstalling): array
```
Comment (live627) [inline fix_packages.php:215]:
```suggestion
	public static function list_getPacks(int $start, int $items_per_page, string $sort, bool $isInstalling): array
```

also, why are these two methods static?

Comment (jdarwood007) [reply fix_packages.php:215]: Honestly can't remember.  I wrapped it in a class less for the object and more for ensuring the functions are local to the tool.  We can do that though if it needs done.

**Pull request merged**